presto-main/src/main/java/com/facebook/presto/translator/AbstractExpressionTranslator.java

```diff
@@ -0,0 +1,129 @@
+package com.facebook.presto.translator;
+
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.InputReferenceExpression;
+import com.facebook.presto.spi.relation.LambdaDefinitionExpression;
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.spi.translator.ExpressionTranslator;
+import com.facebook.presto.spi.translator.TargetExpression;
+import com.facebook.presto.translator.registry.ConstantTypeRegistry;
+import com.facebook.presto.translator.registry.FunctionRegistry;
+
+import java.util.Map;
+
+public abstract class AbstractExpressionTranslator<T>
+        implements ExpressionTranslator
+{
+    private final FunctionRegistry functionRegistry;
+    private final ConstantTypeRegistry constantTypeRegistry;
+
+    public AbstractExpressionTranslator(FunctionRegistry functionRegistry, ConstantTypeRegistry constantTypeRegistry)
+    {
+        this.functionRegistry = functionRegistry;
+        this.constantTypeRegistry = constantTypeRegistry;
+    }
+
+    @Override
+    public TargetExpression<T> translateCallExpression(CallExpression callExpression, Map context)
+    {
+        return null;
+    }
+
+    @Override
+    public TargetExpression<T> translateInputReferenceExpression(InputReferenceExpression inputReferenceExpression, Map context)
+    {
+        return null;
+    }
+
+    @Override
+    public TargetExpression<T> translateConstantExpression(ConstantExpression constantExpression, Map context)
```

Comment:
Mocked functions, will implement once we get consensus on how this interface *should* look.